pkg/helm/actions/install_chart.go

```diff
@@ -45,6 +141,34 @@ func InstallChart(ns, name, url string, vals map[string]interface{}, conf *actio
 	if ch.Metadata.Name != "" && ch.Metadata.Version != "" {
 		metrics.HandleconsoleHelmInstallsTotal(ch.Metadata.Name, ch.Metadata.Version)
 	}
-
+	// remove all the tls related files created by this process
+	defer func() {
+		if os.Getenv("HELM_CLEANUP") == "0" {
+			for _, f := range tlsFiles {
+				fmt.Println(f.Name())
+			}
+			return
+		}
+		for _, f := range tlsFiles {
+			os.Remove(f.Name())
+		}
+	}()
 	return release, nil
 }
+
+func getRepoConnectionConfig(repoName, repoNamespace string, client dynamic.Interface) (*v1beta1.ConnectionConfig, error) {
+	var err error
+	var helmRepoUnstructured *unstructured.Unstructured
+	helmRepoUnstructured, err = client.Resource(helmChartRepositoryNamespaceGVK).Namespace(repoNamespace).Get(context.TODO(), repoName, v1.GetOptions{})
+	if err != nil {
+		helmRepoUnstructured, err = client.Resource(helmChartRepositoryClusterGVK).Get(context.TODO(), repoName, v1.GetOptions{})
+		if err != nil {
+			klog.Errorf("Error listing namespace helm chart repositories: %v \nempty repository list will be used", err)
+			return nil, err
+		}
+	}
+	var helmRepo v1beta1.ProjectHelmChartRepository
+	err = runtime.DefaultUnstructuredConverter.
+		FromUnstructured(helmRepoUnstructured.Object, &helmRepo)
```

Comment:
This conversion should happen only if line 162 doesn't produce an error.

Something along the lines of:

```go
	helmRepoUnstructured, err = client.Resource(helmChartRepositoryNamespaceGVK).Namespace(repoNamespace).Get(context.TODO(), repoName, v1.GetOptions{})
	if err == nil { // conversion happens when no error is produced
		var helmRepo v1beta1.ProjectHelmChartRepository
		err = runtime.DefaultUnstructuredConverter.FromUnstructured(helmRepoUnstructured.Object, &helmRepo)
		return &helmRepo.Spec.Connection.Config, nil
    }
```

Similar construct would apply for the cluster wide chart repository resource query.